spec/unit/use_cases/save_items_details_spec.rb

```diff
@@ -1,64 +1,46 @@
 # frozen_string_literal: true
 
+class DummyItem
+end
 describe SaveItemsDetails do
-  let(:items_gateway) { spy }
+  let(:items_gateway) { spy(all: []) }
   let(:use_case) do
     described_class.new(items_gateway: items_gateway)
   end
 
   it 'uses the items gateway to save the order details' do
-    use_case.execute(items: [{
+    use_case.execute(item: {
                        id: '456',
                        name: 'Bots',
                        price: '10.00',
                        quantity: '2'
-                     }])
-    expect(items_gateway).to have_received(:save) do |items|
-      expect(items[0].id).to eq('456')
-      expect(items[0].name).to eq('Bots')
-      expect(items[0].price).to eq('10.00')
-      expect(items[0].quantity).to eq('2')
+                     })
+    expect(items_gateway).to have_received(:save) do |item|
+      expect(item[0].id).to eq('456')
+      expect(item[0].name).to eq('Bots')
+      expect(item[0].price).to eq('10.00')
+      expect(item[0].quantity).to eq('2')
     end
   end
 
   it 'uses the items gateway to save multiple rows' do
-    items = [
-      { id: '999', name: 'Bobs', price: '99.00', quantity: '25' },
-      { id: '167', name: 'Lobs', price: '145.00', quantity: '3' }
-    ]
-    use_case.execute(items: items)
-    expect(items_gateway).to have_received(:save) do |items|
-      expect(items[0].id).to eq('999')
-      expect(items[0].name).to eq('Bobs')
-      expect(items[0].price).to eq('99.00')
-      expect(items[0].quantity).to eq('25')
-      expect(items[1].id).to eq('167')
-      expect(items[1].name).to eq('Lobs')
-      expect(items[1].price).to eq('145.00')
-      expect(items[1].quantity).to eq('3')
+    gateway = spy(all: [DummyItem.new])
+    item_use_case = described_class.new(items_gateway: gateway)
+    item2 = { id: '167', name: 'Lobs', price: '145.00', quantity: '3' }
+    item_use_case.execute(item: item2)
```

Comment:
change this to item instead of item2

Reply:
Also do we need the DummyItem class?

Reply:
thats how we simulate an item already existing in the file_item_gateway. Without it we would just be adding an item to an empty array again.

Reply:
Instead of creating empty class in the spec file.  can do inside the describe block``` let(:dummy_item) { Class.new } ``` ?